build_tools/checks.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+repo_root="$(dirname "$0")/.."
+
+"$repo_root/build_tools/style.fish" --all --check
+
+cargo build --workspace --all-targets
```

Comment:
Unrelated: I think `printf/Cargo.toml` should defer to dependency versions from the workspace `Cargo.toml`, to remove some duplication.